src/main/java/org/opensearch/knn/index/mapper/LegacyFieldMapper.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.mapper;
+
+import lombok.extern.log4j.Log4j2;
+import org.apache.lucene.document.FieldType;
+import org.opensearch.common.Explicit;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.mapper.ParametrizedFieldMapper;
+import org.opensearch.knn.index.KNNSettings;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import static org.opensearch.knn.common.KNNConstants.DIMENSION;
+import static org.opensearch.knn.common.KNNConstants.HNSW_ALGO_EF_CONSTRUCTION;
+import static org.opensearch.knn.common.KNNConstants.HNSW_ALGO_M;
+import static org.opensearch.knn.common.KNNConstants.KNN_ENGINE;
+import static org.opensearch.knn.common.KNNConstants.METHOD_PARAMETER_SPACE_TYPE;
+import static org.opensearch.knn.common.KNNConstants.SPACE_TYPE;
+
+/**
+ * Field mapper for original implementation
```

Comment:
minor: Do you think we should add what the mapping looks like so users have more context?
```
{
   "type": "knn_vector",
   "dimension": 128
}
```

Also, add something like "This is the mapper for the original version. It defaults to using nmslib as the engine and retrieves parameters from index settings."

Reply:
ack